### ISSUE TITLE
Menu / MenuITem doesn't close after using keyboard navigation to select the option

### ISSUE THREAD

**Issue by dawidzarzycki** [Sage/carbon#4792]
### Current behaviour

When I'm trying to use keyboard navigation to navigate through the menu and then when I use `enter` to select some menu item then I can see that menu stays open

### Expected behaviour

When I press `enter` on the menu item I would like to see the menu closes itself

### CodeSandbox or Storybook URL

https://carbon.sage.com/iframe.html?id=menu--default-story&viewMode=docs#

### JIRA Ticket (Sage Only)

SBS-27087

### Carbon Version

102.19.1

### What browsers are you seeing the problem on?

Chrome

### What Operating System are you seeing the problem on?

MacOS

### Anything else we should know?

I'm working on the SkipLink in our application. The first element on the page after pressing tab when user uses keyboard to navigate through the site it should be SkipLink to allow users jump to main content instead of going through the whole menu. But for now after selecting any menu item with keyboard nav by pressing enter the menu stays open and focus stays on the element I choose. So after pressing enter then pressing tab the next component is not SkipLink but the next button in menu order. I reached the point that after picking any menu item with enter and then pressing tab the focus land on the SkipLink. But even though the focus goes in the correct place the menu stays open. This is why I reporting the bug

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by nicktitchmarsh** [comment]
Hi @dawidzarzycki, if you are focused on a menu item, you should be navigating to another page, in which case the menu would be re-rendered. The skiplink should be the first link on any page, so if you are focused on the menu, you are already past the skiplink. Do you have a demo video of the issue?
